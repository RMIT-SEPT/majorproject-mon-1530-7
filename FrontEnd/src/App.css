.App {
  text-align: center;
  background-color: rgba(164, 214, 197, 0.904);
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.page-container{
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.content-wrap{
  flex:1;
}