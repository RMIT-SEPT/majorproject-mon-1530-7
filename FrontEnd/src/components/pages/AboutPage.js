import React, { Component } from 'react'
import 'bootstrap/dist/css/bootstrap.css';
import { Link } from 'react-router-dom';
import { Col, Container, Jumbotron, Row } from 'react-bootstrap';
import Image4 from '../images/img_4.png';

class AboutPage extends Component {
    render() {
        return (
            <Jumbotron id="jumbotron-custom">
                <Container>
                    <Row>
                        <Col id="col-main4">
                            <img src={Image4} alt="main_img" height="600"></img>
                        </Col>
                        <Col id="col-main5">
                            <h2 className="h2-main">About Us</h2>
                            <p>We pride ourselves on our simple and easy-to-use API for all our
                                clients. Our API will allow your business to thrive and grow further,
                                allowing your to manage everything in one place. From bookings to employees,
                                our API has everything under control for you. Our stress-free user interface
                                will provide all the necessary resources and tools for you anywhere, anytime.
                            </p>

                            <h4 className="h5-main2">Want to find out more?</h4>
                            <p>Our large range of features will be sure to boost your business as
                                well as improve productivity for you and your brand. Keep on track
                                with your bookings and manage everything with our API.
                            </p>
                            <Link to="/signup"><button className="btn-filled" id="align-btn">Get Started</button></Link>
                        </Col>
                    </Row>
                </Container>
            </Jumbotron>
        );
    }
}

export default AboutPage;
