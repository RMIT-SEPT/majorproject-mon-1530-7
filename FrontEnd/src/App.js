import React from "react";
import { Redirect } from 'react-router';
import NavigationBar from "./components/layouts/Navbar.js";
import Footer from "./components/layouts/Footer.js";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import LoginPage from "./components/pages/LoginPage";
import HomePage from "./components/pages/HomePage";
import AboutPage from "./components/pages/AboutPage";
import ContactPage from "./components/pages/ContactPage";
import CustomerBookingPage from "./components/pages/CustomerBookingPage.js";
import NewEmployeePage from "./components/pages/NewEmployeePage.js";

import AdminPastBooking from "./components/pages/AdminPastBooking.js";
import CustomerPastBookingsPage from "./components/pages/CustomerPastBookingsPage.js";

import Dashboard from "./components/pages/Dashboard.js";
import AdminDashboard from "./components/pages/AdminDashboard.js";
import Account from "./components/pages/Account.js";
import Profile from "./components/pages/Profile.js";

import ManageEmp from "./components/pages/ManageEmp.js";
import EmpDetails from "./components/pages/EmpDetails.js";

import UserProfile from "./UserProfile.js"
import AdminPastBookingsPage from "./components/pages/AdminPastBookingsPage.js";

function App() {
  console.log(UserProfile.getLoggedIn())
  if(UserProfile.getLoggedIn()) {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
                <Route exact path="/" component={HomePage} />,
                <Redirect from="/login" to="dashboard" />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Route path="/dashboard" component={Dashboard} >
                  {UserProfile.getAdmin() ? <Redirect to="/admin-dashboard" /> : <Dashboard />}
                </Route>,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Route path="/account" component={Account} >
                  {UserProfile.getLoggedIn() ? <Account /> : <Redirect to="/login"/>}
                </Route>,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} /> 
          
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
  else {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
              
                <Route exact path="/" component={HomePage} />,
                <Route path="/login" component={LoginPage} />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Redirect from="/dashboard" to="/login" />,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Redirect from="/account" to="/login" />,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} />  
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
}
 

export default App;
