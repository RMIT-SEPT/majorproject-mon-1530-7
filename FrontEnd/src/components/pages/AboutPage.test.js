import React from "react";
import { MemoryRouter } from "react-router";
import AboutPage from "./AboutPage";
import App from "../../App";
import { mount, shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Testing AboutPage route", () => {
  it("Ensures AboutPage cotains the correct information from the coresponding route", () => {
    const component = mount(
      <MemoryRouter initialEntries={["/about"]}>
        <App />
      </MemoryRouter>
    );
    expect(component.contains(AboutPage));
  });

  it("renders correctly", () => {
    const wrapper = shallow(<AboutPage />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    shallow(<AboutPage />);
  });
});
