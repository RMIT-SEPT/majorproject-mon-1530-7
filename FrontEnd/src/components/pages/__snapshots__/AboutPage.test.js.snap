// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AboutPage route renders correctly 1`] = `
<Jumbotron
  fluid={false}
  id="jumbotron-custom"
>
  <Container
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        id="col-main4"
      >
        <img
          alt="main_img"
          height="600"
          src="img_4.png"
        />
      </Col>
      <Col
        id="col-main5"
      >
        <h2
          className="h2-main"
        >
          About Us
        </h2>
        <p>
          We pride ourselves on our simple and easy-to-use API for all our clients. Our API will allow your business to thrive and grow further, allowing your to manage everything in one place. From bookings to employees, our API has everything under control for you. Our stress-free user interface will provide all the necessary resources and tools for you anywhere, anytime.
        </p>
        <h4
          className="h5-main2"
        >
          Want to find out more?
        </h4>
        <p>
          Our large range of features will be sure to boost your business as well as improve productivity for you and your brand. Keep on track with your bookings and manage everything with our API.
        </p>
        <Link
          to="/signup"
        >
          <button
            className="btn-filled"
            id="align-btn"
          >
            Get Started
          </button>
        </Link>
      </Col>
    </Row>
  </Container>
</Jumbotron>
`;
